tests/lib/nsxv_test.py

```diff
@@ -747,108 +794,156 @@ def testBadHeaderCase4(self):
+    self.verifyAddressType(dest_addr, 'Ipv4Address')
     self.naming.GetNetAddr.assert_has_calls(
         [mock.call('INTERNAL'), mock.call('GOOGLE_DNS')])
 
-  def testNsxvStrForSourceIpv4Mismatch(self):
+  def testMixedToV6(self):
+     self.naming.GetNetAddr.side_effect = [
+         [nacaddr.IP('8.8.4.4'), nacaddr.IP('8.8.8.8'),
+          nacaddr.IP('2001:4860:4860::8844'),
+          nacaddr.IP('2001:4860:4860::8888')],
+         [nacaddr.IP('2001:4860:8000::/33')]]
+
+     source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                          MIXED_TO_V6)
+     self.verifyAddressType(source_addr, 'Ipv6Address')
+     self.verifyAddressType(dest_addr, 'Ipv6Address')
+     self.naming.GetNetAddr.assert_has_calls(
+         [mock.call('GOOGLE_DNS'), mock.call('SOME_HOST')])
+
+  def testV6ToMixed(self):
     self.naming.GetNetAddr.side_effect = [
         [nacaddr.IP('2001:4860:8000::/33')],
-        [nacaddr.IP('8.8.4.4'),
-         nacaddr.IP('8.8.8.8'),
+        [nacaddr.IP('8.8.4.4'), nacaddr.IP('8.8.8.8'),
          nacaddr.IP('2001:4860:4860::8844'),
          nacaddr.IP('2001:4860:4860::8888')]]
-    pol = policy.ParsePolicy(MIXED_HEADER + SOURCE_IPv4_MISMATCH, self.naming,
-                             False)
-    target = nsxv.Nsxv(pol, EXP_INFO)
 
-    # parse the output and seperate sections and comment
-    section_tokens = str(target).split('<section')
-    sections = []
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                         V6_TO_MIXED)
+    self.verifyAddressType(source_addr, 'Ipv6Address')
+    self.verifyAddressType(dest_addr, 'Ipv6Address')
+    self.naming.GetNetAddr.assert_has_calls(
+        [mock.call('SOME_HOST'), mock.call('GOOGLE_DNS')])
 
-    for sec in section_tokens:
-      section = sec.replace('name=', '<section name=')
-      sections.append(section)
-    # parse the xml
-    root = ET.fromstring(sections[1])
+  def testMixedToMixed(self):
+    self.naming.GetNetAddr.side_effect = [
+        [nacaddr.IP('8.8.4.4'), nacaddr.IP('8.8.8.8'),
+         nacaddr.IP('2001:4860:4860::8844'),
+         nacaddr.IP('2001:4860:4860::8888')],
+        [nacaddr.IP('8.8.4.4'), nacaddr.IP('8.8.8.8'),
+         nacaddr.IP('2001:4860:4860::8844'),
+         nacaddr.IP('2001:4860:4860::8888')]]
 
-    # check source has only Ipv6
-    source_address = root.findall('./rule/sources/source')
-    self.assertIsNotNone(source_address)
-    for source_addr in source_address:
-      addr_type = source_addr.find('type').text
-      self.assertEquals(addr_type, 'Ipv6Address')
+    source_addr, dest_addr = self.getSourceDestAddresses(MIXED_HEADER +
+                                                         MIXED_TO_MIXED)
+    self.assertFalse(len(source_addr) == 0)
+    for saddr in source_addr:
+      saddr_type = saddr.find('type').text
+      self.assertIn(saddr_type, ('Ipv4Address', 'Ipv6Address'))
```

Comment:
So I think we might have a problem here. Is this ever not going to be true? Its always going to be one or the other right (unless the term is missing). This doesn't check that both exist which I think is the true point of this test.